include/ensmallen_bits/age_moea/agemoea_impl.hpp

```diff
@@ -0,0 +1,781 @@
+        distancesTemp[i] = arma::accu(arma::pow((pa - t * ba), 2));
+  }
+  distances = distancesTemp;
+}
+
+//! Sort population into Pareto fronts.
+template<typename MatType>
+inline void AGEMOEA::FastNonDominatedSort(
+    std::vector<std::vector<size_t> >& fronts,
+    std::vector<size_t>& ranks,
+    std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives)
+{
+  std::map<size_t, size_t> dominationCount;
+  std::map<size_t, std::set<size_t> > dominated;
+
+  // Reset and initialize fronts.
+  fronts.clear();
+  fronts.push_back(std::vector<size_t>());
+
+  for (size_t p = 0; p < populationSize; p++)
+  {
+    dominated[p] = std::set<size_t>();
+    dominationCount[p] = 0;
+
+    for (size_t q = 0; q < populationSize; q++)
+    {
+      if (Dominates<MatType>(calculatedObjectives, p, q))
+        dominated[p].insert(q);
+      else if (Dominates<MatType>(calculatedObjectives, q, p))
+        dominationCount[p] += 1;
+    }
+
+    if (dominationCount[p] == 0)
+    {
+      ranks[p] = 0;
+      fronts[0].push_back(p);
+    }
+  }
+
+  size_t i = 0;
+
+  while (!fronts[i].empty())
+  {
+    std::vector<size_t> nextFront;
+
+    for (size_t p: fronts[i])
+    {
+      for (size_t q: dominated[p])
+      {
+        dominationCount[q]--;
+
+        if (dominationCount[q] == 0)
+        {
+          ranks[q] = i + 1;
+          nextFront.push_back(q);
+        }
+      }
+    }
+
+    i++;
+    fronts.push_back(nextFront);
+  }
+  // Remove the empty final set.
+  fronts.pop_back();
+}
+
+//! Check if a candidate Pareto dominates another candidate.
+template<typename MatType>
+inline bool AGEMOEA::Dominates(
+    std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives,
+    size_t candidateP,
+    size_t candidateQ)
+{
+  bool allBetterOrEqual = true;
+  bool atleastOneBetter = false;
+  size_t n_objectives = calculatedObjectives[0].n_elem;
+
+  for (size_t i = 0; i < n_objectives; i++)
+  {
+    // P is worse than Q for the i-th objective function.
+    if (calculatedObjectives[candidateP](i) > calculatedObjectives[candidateQ](i))
+      allBetterOrEqual = false;
+
+    // P is better than Q for the i-th objective function.
+    else if (calculatedObjectives[candidateP](i) < calculatedObjectives[candidateQ](i))
+      atleastOneBetter = true;
+  }
+
+  return allBetterOrEqual && atleastOneBetter;
+}
+
+//! Assign diversity score for a given point and teh selected set.
+template <typename MatType>
+inline typename MatType::elem_type AGEMOEA::DiversityScore(std::set<size_t>& selected,
+                                                  const MatType& pairwiseDistance,
+                                                  size_t S)
+{ 
+  typedef typename MatType::elem_type ElemType;
+  ElemType m = arma::datum::inf;
+  ElemType m1 = arma::datum::inf;
+  std::set<size_t>::iterator it;
+  for (it = selected.begin(); it != selected.end(); it++)
+  {
+    if(*it == S){ continue; }
```

Comment:
```suggestion
    if (*it == S)
      continue;
```
Here is a link to the style guide, if you don't mind going through the whole code, I haven't checked everything in this file (but skimming through I can see there are a lot of places where it doesn't look quite right): https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines